src/domain/user/create-user.use-case.ts

```diff
@@ -1,11 +1,36 @@
 import bcrypt from 'bcrypt';
 
-import { AppDataSource } from '../../data-source.js';
-import { User } from '../../data/db/entity/user.entity.js';
-import { emailValidator, passwordValidator } from '../../data/validators/validators.js';
-import { UserInput } from '../../model/user.model.js';
-import { CustomError } from '../../format-error.js';
-import { authorizeToken } from '../../auth/authorize-token.js';
+import { AppDataSource } from '../../data-source';
+import { User } from '../../data/db/entity/user.entity';
+import { emailValidator, passwordValidator } from '../../data/validators/validators';
+import { UserInput } from '../../model/user.model';
+import { CustomError } from '../../format-error';
+import { authorizeToken } from '../../auth/authorize-token';
+
+export class CreateUserUseCase {
+  constructor(private readonly repository = AppDataSource.getRepository(User), private readonly user = new User()) {}
```

Comment:
mesam coisa aqui. Mas o caso do user, pode criar ele inline mesmo, porque o ideial é sempre criar uma nova instancia para ser populada pelos campos do input, certo? Se definir na instancia da classe, a classe User se manterá entre as requisicoes (considerando que no futuro teremos uma só intancia da CreateUserUseCase)

```suggestion
  private readonly repository = AppDataSource.getRepository(User)
```